src/garage/tf/misc/tensor_utils.py

```diff
@@ -48,32 +53,50 @@ def get_target_ops(variables, target_variables, tau=None):
 
 
 def flatten_batch(t, name='flatten_batch'):
+    """Flatten batch."""
     return tf.reshape(t, [-1] + list(t.shape[2:]), name=name)
 
 
 def flatten_batch_dict(d, name=None):
+    """Flatten batch dict."""
     with tf.name_scope(name, 'flatten_batch_dict', [d]):
         return {k: flatten_batch(v) for k, v in d.items()}
 
 
+def flatten_input_along_dim(t, dim, name='flatten_input_full'):
```

Comment:
how is this different from flatten_batch?

please include a full docstring

Reply:
Will do.
In short, `flatten_batch` flattens the batch dimension while keeping the sample dimension, and this function is able to flatten the observation dimension, when the given `dim` is 2 (since we have [batch_dim, time_step, observation_dim] for samples in garage).